### PR TITLE
feat(deploy): Use HMLanguageTools for LOCR/RTLV

### PR DESCRIPTION
**Pull request by AnthonyFuller**
This PR does many a thing:
- Adds `locr.json` ~~and `rtlv.json`~~
- Reworks `localisation` and `localisationOverrides` to use HMLanguageTools
- Updates some action workflow dependencies to resolve warnings (and to prevent them breaking due to deprecation)

Eventually, RPKG will use HMLanguageTools so `-extract_locr_as_json_from` can be restored simply instead of having to extract the raw files and convert them, which will make this much cleaner.

### CODE REVIEW

**Comment by AnthonyFuller** [comment]
**DO NOT MERGE, RTLV IS CURRENTLY BROKEN DUE TO A RESOURCETOOL ISSUE**

**Comment by atampy25** [comment]
Does not LGTM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     if: startsWith(github.ref, 'refs/tags/')
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (atampy25) [inline .github/workflows/release.yml:14]: LGTM

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -21,14 +21,14 @@ jobs:
 
       - name: Cache
         id: cache
-        uses: actions/cache@v2.1.6
+        uses: actions/cache@v3
```
Comment (atampy25) [inline .github/workflows/release.yml:24]: LGTM

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -21,14 +21,14 @@ jobs:
...
         with:
           path: node_modules
           key: buildCachev24
 
       - name: Cache
         id: cache2
-        uses: actions/cache@v2.1.6
+        uses: actions/cache@v3
```
Comment (atampy25) [inline .github/workflows/release.yml:31]: LGTM

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -134,7 +134,7 @@ jobs:
 
       - name: Cache
         id: cache
-        uses: actions/cache@v2.1.6
+        uses: actions/cache@v3
```
Comment (atampy25) [inline .github/workflows/release.yml:137]: LGTM

File: src/deploy.ts [hunk 5]
```diff
@@ -2166,33 +2328,42 @@ export default async function deploy(
...
+			process.cwd(),
+			"staging",
+			localisationFileRPKG.replace(/patch[0-9]*/gi, ""),
+			"00F5817876E691F1.LOCR"
+		)}" --metapath "${path.join(
+			process.cwd(),
+			"staging",
+			localisationFileRPKG.replace(/patch[0-9]*/gi, ""),
+			"00F5817876E691F1.LOCR.meta.json"
+		)}"`)
```
Comment (atampy25) [inline src/deploy.ts:2366]: This all better work

File: docs/Special File Types.md [hunk 6]
```diff
@@ -140,4 +146,4 @@ The framework will automatically extract the WWEV and copy the wem to it, then r
 
 Patches a given file with a VCDiff delta using xdelta3. Named in the format `hash~filetype.delta`. For example, to patch `004A314BA75429D7.GFXF`, the file should be named `004A314BA75429D7~GFXF.delta`.
 
-The framework will automatically extract the file and patch it, then place it in the staging folder. You may want to include a meta file for the hash as well, or the RPKG tool will assume the default metadata.
\ No newline at end of file
+The framework will automatically extract the file and patch it, then place it in the staging folder. You may want to include a meta file for the hash as well, or the RPKG tool will assume the default metadata.
```
[the remark this answers is not in the document]
Comment (AnthonyFuller) [reply docs/Special File Types.md:149]: Absolutely nothing, I didn't touch it.

File: src/deploy.ts (hunk 5 above)
Comment (AnthonyFuller) [reply src/deploy.ts:2366]: It does, I've tested it.

File: src/types.ts [hunk 7]
```diff
@@ -466,3 +466,12 @@ export interface ModAPI {
...
 }
+
+export interface TonyToolsLOCR {
+	hash: string
+	languages: {
+		[lang: string]: {
+			[hash: string]: string
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (AnthonyFuller) [reply src/types.ts:477]: It does, I've tested it.

**Comment by AnthonyFuller** [comment]
@atampy25 I have removed RTLV support (by commenting it out) for now. I'd like to get this merged.

**Pull request merged**